lib/puppet/provider/parsedfile.rb

```diff
@@ -260,12 +274,19 @@ def self.prefetch_all_targets(resources)
 
   # Prefetch an individual target.
   def self.prefetch_target(target)
-
     begin
       target_records = retrieve(target)
     rescue Puppet::Util::FileType::FileReadError => detail
-      puts detail.backtrace if Puppet[:trace]
-      Puppet.err _("Could not prefetch %{resource} provider '%{name}' target '%{target}': %{detail}. Treating as empty") % { resource: self.resource_type.name, name: self.name, target: target, detail: detail }
+      if ! @swallow_prefetch_errors
```

Comment:
I was suggesting flipping the `if`/`else`, but this works too. I'd read the flipped version as "if we are swallowing prefetch errors, just do the output, else we mark it as failed."

Reply:
Ah I see. I usually like having the `more code` part of the logic be in the else branch which in this case would be the output printing. Makes things easier to read.

Reply:
Yes, that is a better name and it is a worthwhile change. Feel free to put up the rename PR if you'd like, else I can do it later tomorrow morning. Feeling lazy right now.

Reply:
Rename PR was here: https://github.com/puppetlabs/puppet/pull/7190